app/controllers/spree/api/images_controller_decorator.rb

```diff
@@ -0,0 +1,15 @@
+Spree::Api::ImagesController.class_eval do
+  def update_product_image
+    @product = Spree::Product.find(params[:product_id])
+    authorize! :update, @product
+
+    if @product.images.first.nil?
+      @image = Spree::Image.create(attachment: params[:file], viewable_id: @product.master.id, viewable_type: 'Spree::Variant')
+      respond_with(@image, :status => 201)
+    else
+      @image = @product.master.images.first
+      @image.update_attributes(attachment: params[:file])
+      respond_with(@image, :status => 200)
```

Comment:
I'd use symbol syntax instead of hash rockets. I wonder why rubocop does not complain about it :thinking: